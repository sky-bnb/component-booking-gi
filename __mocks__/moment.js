const date = '2019-11-08T19:28:54.467Z';

const moment = jest.fn(() => ({
  startOf: () => ({ format: () => 1 }),
  endOf: () => ({ format: () => 31 }),
  format: () => '2019-11',
  add: () => ({ format: () => '2019-09-11' })
}));

module.exports = moment;
