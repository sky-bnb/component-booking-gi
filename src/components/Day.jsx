import React from 'react';
import styled from 'styled-components';
import moment from 'moment';

export const StyledDay = styled.td`
  font-size: 14px;
  width: 37px;
  height: 37px;
  display: table-cell;
  font-family: 'Quicksand', sans-serif;
  font-weight: 700;
  line-height: 12px;
  text-align: center;
  text-decoration: ${({ isActive }) => isActive ? 'none' : 'line-through'};
  color: ${({ isActive }) => isActive ? 'rgb(72, 72, 72)' : 'rgb(216, 216, 216)'};
  border: ${({ isBlank }) => isBlank ? 'none' : '1px double rgb(228, 231, 231)'};
  border-spacing: 0px;
  padding: 1px;
  margin: 0px;
  ${({ isSelected }) => isSelected ? 'background: rgb(0, 166, 153); color: rgb(255, 255, 255); text-decoration: none;' : null}

  ${({ isActive, isSelected }) => isActive  && ! isSelected ? ':hover { background: rgb(228, 231, 231); }' : null}
`;

const Day = (props) => {
  const { available, calendarMonth, date, isBlank, isCheckinOpen, isCheckoutOpen, updateCheckinDate, updateCheckoutDate, checkinValue, checkoutValue } = props;
  const displayDate = (value) => moment(value).format('MM/DD/YYYY').toString();
  const dateString = date ? moment(calendarMonth).format('YYYY-MM').toString().concat('-', date.toString().padStart(2, '0')) : null;
  const isCheckinAvail = available.includes(dateString);
  const isCheckoutAvail = available.includes(moment(dateString).add(1, 'days').format('YYYY-MM-DD').toString());
  const isActive = (isCheckinAvail && isCheckinOpen) || (isCheckoutAvail && isCheckoutOpen);
  const isSelected = displayDate(dateString) === checkinValue || displayDate(dateString) === checkoutValue;
  const updateDate = (date) => isCheckinOpen ? updateCheckinDate({target: {value: date}}) : updateCheckoutDate({target: {value: date}});

  return (
    <StyledDay value={dateString} isActive={isActive} isSelected={isSelected} isBlank={isBlank} checkinValue={checkinValue} checkoutValue={checkoutValue} onClick={() => updateDate(moment(dateString).format('MM/DD/YYYY').toString())}>
      {date}
    </StyledDay>
  );
};

export default Day;
