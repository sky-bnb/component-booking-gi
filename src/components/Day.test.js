import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import moment from 'moment';

import Day, { StyledDay } from './Day.jsx';

const {
  available,
  calendarMonth,
  date,
  index,
  isBlank,
  isCheckinOpen,
  isCheckoutOpen,
  checkinValue,
  checkoutValue,
  updateCheckinDate,
  updateCheckoutDate,
} = {
  available: ['2019-06-02', '2019-07-01'],
  calendarMonth: moment(),
  date: 1,
  key: 1,
  isBlank: false,
  isCheckinOpen: false,
  checkinValue: '',
  isCheckoutOpen: false,
  checkoutValue: '',
  updateCheckinDate: () => null,
  updateCheckoutDate: () => null,
};

describe('Day Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Day available={available}
    calendarMonth={calendarMonth}
    date={date}
    key={index}
    isBlank={isBlank}
    isCheckinOpen={isCheckinOpen}
    isCheckoutOpen={isCheckoutOpen}
    checkinValue={checkinValue}
    checkoutValue={checkoutValue}
    updateCheckinDate={updateCheckinDate}
    updateCheckoutDate={updateCheckoutDate}
  />));

  it('renders the Day component', () => {
    expect(wrapper.find(StyledDay).length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
