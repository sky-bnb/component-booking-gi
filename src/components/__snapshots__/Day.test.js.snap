// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Component matches the snapshot 1`] = `
<Day__StyledDay
  checkinValue=""
  checkoutValue=""
  isActive={false}
  isBlank={false}
  isSelected={false}
  onClick={[Function]}
  value="2019-11-01"
>
  1
</Day__StyledDay>
`;
